macaw-riscv/src/Data/Macaw/RISCV/Identify.hs

```diff
@@ -62,7 +62,12 @@ matchReturn :: RISCVConstraints rv
             -> Maybe (Some (MA.AbsValue w))
 matchReturn rvRepr absProcState ip = G.withRV rvRepr $ do
   MC.BVAnd _ addr (MC.BVValue _ mask) <- MC.valueAsApp ip
-  guard (mask == 0xfffffffffffffffe)
+  let maskVal = case G.rvBaseArch rvRepr of
+                  G.RV64Repr -> 0xfffffffffffffffe
+                  G.RV32Repr -> 0xfffffffe
+                  _ -> error ( "Unsupported base architecture: "
+                            ++ (show (G.rvBaseArch rvRepr)))
```

Comment:
What do you think about converting the mask into a `BV w` from bv-sized and then making `maskVal = sext w (mkBV (knownNat @8) 0xfe)`

That would be resilient to any architecture width.

Reply:
Done!